***
# <font color=green size=10>CURSO DE ESTADÍSTICA - PARTE 2</font>
***

# <font color=green>1 CONOCIENDO LOS DATOS</font>
***

## <font color=green>1.1 Dataset del proyecto</font>
***

### Encuesta Nacional por Muestreo de Hogares - 2015

La **Encuesta Nacional por Muestreo de Hogares - ENMH** en Brasil anualmente investiga, de manera permanente, características generales de la población, educación, trabajo, ingresos, vivienda y otras, con frecuencia variables, de acuerdo a las necesidades de información del país, tales como las características de migración, fecundidad, nupcialidad, salud, seguridad alimentaria, entre otros temas. La recopilación de estas estadísticas constituye, durante los 49 años de realización de la investigación, un importante instrumento para la formulación, validación y evaluación de políticas orientadas al desarrollo socioeconómico y la mejora de las condiciones de vida en Brasil.

### Fuente de los Datos

Instituto Brasileño de Geografía y Estadística (IBGE) Encuesta Nacional por Muestreo de Hogares(PNAD) en el 2015

### Variables utilizadas

> ### Ingresos
> ***

Ingresos mensuales del trabajo principal para personas de 10 años o más.

> ### Edad
> ***

Edad del residente en años en la fecha de referencia .

> ### Altura (elaboración propia)
> ***

Altura del habitante en metros.

> ### UF
> ***

Unidad Federal

|Código|Descripción|
|---|---|
|11|Rondônia|
|12|Acre|
|13|Amazonas|
|14|Roraima|
|15|Pará|
|16|Amapá|
|17|Tocantins|
|21|Maranhão|
|22|Piauí|
|23|Ceará|
|24|Rio Grande do Norte|
|25|Paraíba|
|26|Pernambuco|
|27|Alagoas|
|28|Sergipe|
|29|Bahia|
|31|Minas Gerais|
|32|Espírito Santo|
|33|Rio de Janeiro|
|35|São Paulo|
|41|Paraná|
|42|Santa Catarina|
|43|Rio Grande do Sul|
|50|Mato Grosso do Sul|
|51|Mato Grosso|
|52|Goiás|
|53|Distrito Federal|

> ### Sexo
> ***

|Código|Descripción|
|---|---|
|0|Masculino|
|1|Femenino|

> ### Años de Estudio
> ***

|Código|Descripción|
|---|---|
|1|Sin educación o menos de 1 año|
|2|1 año|
|3|2 años|
|4|3 años|
|5|4 años|
|6|5 años|
|7|6 años|
|8|7 años|
|9|8 años|
|10|9 años|
|11|10 años|
|12|11 años|
|13|12 años|
|14|13 años|
|15|14 años|
|16|15 años o más|
|17|No determinado|
||No aplica|

> ### Color
> ***

|Código|Descripción|
|---|---|
|0|Indígena|
|2|Blanco|
|4|Negro|
|6|Amarillo|
|8|Pardo|
|9|Sin declarar|

#### <font color='red'>Nota</font>
***
> Se realizaron los siguientes tratamientos sobre los datos originales:
> 1. Se eliminaron los registros en los que **Ingreso** no era válido. (999 999 999 999);
> 2. Se eliminaron los registros en los que **Ingreso** eran missing;
> 3. Solo se consideraron los registros de **Personas de Referencia** de cada hogar (responsable del hogar).

### Importando pandas y leyendo el dataset del proyecto

https://pandas.pydata.org/

In [1]:
import pandas as pd

In [2]:
datos =pd.read_csv('datos.csv', sep=',', encoding='latin-1') #le comando encoding='latin-1 permite detectar letras latinas en las variables como la ñ

In [3]:
datos.head()

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
0      11     0    23      8               12      800  1.603808
1      11     1    23      2               12     1150  1.739790
2      11     1    35      8               15      880  1.760444
3      11     0    46      2                6     3500  1.783158
4      11     1    47      8                9      150  1.690631

Recuerde anotar el tipo de separador de columna que utiliza el archivo CSV. Por defecto, la función read_csv() usa el parámetro sep=','. Si el separador es cualquier carácter que no sea la coma, necesitaremos informar esto para la función, por ejemplo: read_csv('data.csv', sep =';')).

---
---

# <font color=green>2 DISTRIBUCIONES DE PROBABILIDAD</font>
***

## <font color=red>Problema</font>
***

En una competencia para ocupar un puesto de científico de datos, tenemos un total de **10 preguntas** de opción múltiple con **3 alternativas posibles** en cada pregunta. **Cada pregunta tiene el mismo valor.** Suponga que un candidato decide aventurarse sin haber estudiado nada. Decide hacer la prueba con los ojos vendados y adivinar todas las respuestas. Suponiendo que la prueba **vale 10 puntos y la puntuación de corte es 5**, obtenga la probabilidad de que este candidato **acierte 5 preguntas** y también la probabilidad de que el candidato **pase a la siguiente etapa del proceso de selección**.

## <font color=green>2.1 Distribución Binomial</font>
***

Un evento **binomial** se caracteriza por la posibilidad de que solo ocurran dos categorías. Estas categorías juntas representan todo el espacio muestral, siendo también mutuamente excluyentes, es decir, la ocurrencia de una implica la no ocurrencia de la otra.

En el análisis estadístico, el uso más común de la distribución binomial es para resolver problemas que involucran situaciones de **éxito** y **fracaso**.

# $$P(k)=\binom{n}{k} p^k q^{n-k}$$

Donde:

$p$ = probabilidad de éxito

$q = (1 - p)$ = probabilidad de fracaso

$n$ = número de eventos estudiados

$k$ = número de eventos deseados que tienen éxito

### Experimento Binomial

1. Realización de $n$ pruebas idénticas.

2. Las pruebas son independientes.

3. Solo son posibles dos resultados, por ejemplo: Verdadero o falso; Cara o cruz; Éxito o fracaso.

4. La probabilidad de éxito está representada por $p$ y la probabilidad de fracaso por $1-p = q$. Estas probabilidades no cambian de una prueba para otra.

### Media de la distribución binomial

El valor esperado o la media de la distribución binomial es igual al número de experimentos realizados multiplicado por la probabilidad de que ocurra el evento.

# $$\mu = n \times p$$

### Desviación estándar de la distribución binomial

La desviación estándar es la raíz quadrada del producto entre el número de experimentos, la probabilidad de éxito y la probabilidad de fracaso.

# $$\sigma = \sqrt{n \times p \times q}$$

### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.comb.html

In [4]:
from scipy.special import comb

### Combinaciones

El número de combinaciones de $n$ objetos, tomando $k$ cada vez, es:

# $$C_{k}^{n} = \binom{n}{k} = \frac{n!}{k!(n - k)!}$$

Donde

## $$n! = n\times(n-1)\times(n-2)\times...\times(2)\times(1)$$
## $$k! = k\times(k-1)\times(k-2)\times...\times(2)\times(1)$$

Por definición

## $$0! = 1$$

## <font color='blue'>Ejemplo: Mega Sena</font>

En una rueda de lotería Mega Sena tenemos un total de **60 números** para elegir, donde la apuesta mínima es de **seis números**. Usted que es curioso(a) decide calcular la probabilidad de ganar la Mega Sena con solo **un juego**. Para eso necesitamos saber cuántas **combinaciones de seis números se pueden formar con los 60 números disponibles**.

### $$C_{6}^{60} = \binom{60}{6} = \frac{60!}{6!(60 - 6)!}$$

In [5]:
#Número total de enventos posibles de escoger los 6 números de la lotería
combinaciones = comb(60,6)
combinaciones

50063860.0

In [6]:
#La probabilidad de ganar es:
probabilidad = 1 / combinaciones
print(f'{probabilidad:0.15f}')

0.000000019974489


### Ejemplo 2:
Supongamos que acabamos de crear un juego de lotería llamado Exposición de Premios de Alura. En este nuevo juego, el que apuesta marca 20 números, de los 25 disponibles en el boleto, y puede ganar hasta 1 millón de pesos.

Determine el número de combinaciones posibles (espacio muestral) y la probabilidad de ganar el premio jugando solo un boleto (considere sólo quince lugares decimales).

In [7]:
#Número total de enventos posibles de escoger los 25 números de la lotería
combinaciones = comb(25,20)
combinaciones

53130.0

In [8]:
#La probabilidad de ganar es:
probabilidad = 1 / combinaciones
print(f'{probabilidad:0.15f}')

0.000018821757952


## <font color='blue'>Ejemplo: Concurso para científico de datos</font>

En una competencia para ocupar un puesto de científico de datos, tenemos un total de **10 preguntas** de opción múltiple con **3 alternativas posibles** en cada pregunta. **Cada pregunta tiene el mismo valor.** Suponga que un candidato decide aventurarse sin haber estudiado nada. Decide tomar la prueba con los ojos vendados y adivinar todas las respuestas. Suponiendo que la prueba **vale 10 puntos y la puntuación de corte es 5**, obtenga la probabilidad de que este candidato **acierte 5 preguntas** y también la probabilidad de que el candidato **pase a la siguiente etapa del proceso de selección**.

### ¿Cuál es la cantidad de pruebas ($n$)?

In [9]:
#Cantidad de Pruebas
n= 10
n

10

### ¿Las pruebas son independientes?

Sí. La opción elegida en una pregunta no influye en la opción elegida en otra pregunta.

### ¿Sólo dos resultados son posibles en cada prueba?

Sí. El candidato tiene dos posibilidades, ACERTAR o FALLAR una pregunta.

### ¿Cuál es la probabilidad de éxito ($p$)?

In [10]:
numero_de_alternativas_por_pregunta = 3
p= 1/ numero_de_alternativas_por_pregunta #Probabilidad de acertar una pregunta
p

0.3333333333333333

### ¿Cuál es la probabilidad de fracaso ($q$)?

In [11]:
q= 1-p #Probabilidad de fallar una pregunta
q

0.6666666666666667

### ¿Cuál es el número total de eventos para tener éxito ($k$)?

In [12]:
k= 5
k

5

### Solución 1

In [13]:
#probabilidad= comb(n,k)*(p**k)*(q**(n-k))
probabilidad= comb(10,5)*(p**k)*(q**(n-k))
print(f'{probabilidad:.8f}') #Probabilidad de acertar al marcar aleatoriamente las preguntas

0.13656455


### Importando bibliotecas
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.binom.html

En Python, binom es una función de la biblioteca scipy.special que permite calcular coeficientes binomiales, es decir, los valores de
𝐶
(
𝑛
,
𝑘
)
C(n,k) o "n sobre k". Esta función es útil en problemas de combinatoria y probabilidad, donde necesitamos saber el número de formas de elegir
𝑘
k elementos de un conjunto de
𝑛
n elementos sin importar el orden.

La fórmula para el coeficiente binomial es:

# $$C_{k}^{n} = \binom{n}{k} = \frac{n!}{k!(n - k)!}$$



In [14]:
from scipy.stats import binom

### Solución 2

In [15]:
probabilidad = binom.pmf(k,n,p)
print(f'{probabilidad:.8f}')#Probabilidad de acertar al marcar aleatoriamente las preguntas

0.13656455


### Obtenga la probabilidad de que el candidato apruebe

### $$P(acertar \geq 5) = P(5) + P(6) + P(7) + P(8) + P(9) + P(10)$$

In [16]:
 #La probabilidad de acertar mínimo 5 preguntas es:
 binom.pmf(5,n,p) +  binom.pmf(6,n,p) + binom.pmf(7,n,p) +  binom.pmf(8,n,p) +  binom.pmf(9,n,p) +  binom.pmf(10,n,p)

0.21312808006909476

In [17]:
#Forma más eficiente de realizar el cálculo
#La probabilidad de acertar mínimo 5 preguntas es:
binom.pmf([5,6,7,8,9,10],n,p).sum()

0.21312808006909476

In [18]:
# Suma todas las posibilidades que hay hasta 4, es la posibilidad de aprobar el examen vista desde otra perspectiva
1- binom.cdf(4,n,p)

0.21312808006909523

In [19]:
# Otra función más directa es:
binom.sf(4,n,p)

0.21312808006909525

### Ejemplo:
Una moneda perfectamente equilibrada se lanza al aire cuatro veces. Usando la distribución binomial, obtenga la probabilidad de que la moneda caiga con la cruz hacia arriba dos veces.

In [20]:
p = 1 / 2  # Probabilidad de salir CRUZ
n = 4          # Total de lanzamientos
k = 2          # Total de aciertos (CRUZ hacia arriba)

binom.pmf(k, n, p)

0.375

### Ejemplo:
Un dado perfectamente equilibrado se lanza hacia arriba diez veces. Usando la distribución binomial, obtenga la probabilidad de que los dados caigan con el número cinco hacia arriba al menos tres veces.

In [21]:
p = 1 / 6  # Probabilidad de que caiga el número 5 hacia arriba
n = 10     # Total de lanzamientos
k = 2      # Para calcular al menos tres veces, usamos k = 2 en binom.sf

# Calculamos la probabilidad de obtener al menos tres veces el número cinco
probabilidad = binom.sf(k, n, p)

print(f'La probabilidad de obtener el número cinco al menos tres veces es: {probabilidad:.8f}')

La probabilidad de obtener el número cinco al menos tres veces es: 0.22477320


### Media de la Distribución Binomial
## <font color='blue'>Ejemplo: Yincana</font>

Una ciudad del interior realiza una yincana cada año para recaudar fondos para el hospital de la ciudad. En la última competencia se sabe que la **proporción de mujeres participantes fue del 60%**. **El número total de equipos, con 12 miembros, inscritos en la yincana de este año es 30**. Con la información anterior, responda: ¿Cuántos equipos deberían estar formados por **8 mujeres**?

### Solución

In [22]:
p = 0.6 # Proporción de mujeres
p

0.6

In [23]:
n= 12 #cantidad de personas en el equipo
n

12

In [24]:
k = 8 #cantidad de mujeres en el equipo
k

8

In [25]:
probabilidad = binom.pmf(k,n,p)
print(f'{probabilidad:.8f}')#Probabilidad de seleccionar exactamente 8 mujeres en equipos de tamaño 12 donde la probabilidad  de seleccionar una mujer es de 0.6

0.21284094


In [26]:
# De los 30 equipos cuantos de estos pueden estar conformados por 8 mujeres
equipos = 30* probabilidad
equipos.round(0)

6.0

Suponga que la probabilidad de que una pareja tenga hijos con ojos azules es del 22%. En 50 familias, con 3 hijos cada una, ¿cuántas podemos esperar que tengan dos hijos con ojos azules?

In [27]:
p= 0.22 #Probabilidad de tener ojos azules
p
n= 3 #Cantidad de hijos por familia
n
k= 2 #Cantidad de hijos con ojos azules
k
probabilidad = binom.pmf(k,n,p)

# Multiplicamos esta probabilidad por el número de familias para obtener el número esperado de familias
familias_con_dos_hijos_azules = 50 * probabilidad

print(f'Probabilidad de que una familia tenga exactamente 2 hijos con ojos azules: {probabilidad:.8f}')
print(f'Número esperado de familias con exactamente 2 hijos con ojos azules: {familias_con_dos_hijos_azules:.2f}')

Probabilidad de que una familia tenga exactamente 2 hijos con ojos azules: 0.11325600
Número esperado de familias con exactamente 2 hijos con ojos azules: 5.66


In [28]:
#Número esperado de familias con exactamente 2 hijos con ojos azules
p = 0.22
n = 3
k = 2
N = 50

probabilidad = binom.pmf(k, n, p)

media = probabilidad * N
media

5.662799999999996

Lo que aprendimos:

- Los conceptos básicos de la distribución binomial de probabilidad
- Identificar y resolver problemas que involucran un experimento binomial
- Obtener la probabilidad de eventos que involucran sólo dos posibilidades de resultado

---
---

## <font color=red>Problema</font>
***

Un restaurante recibe un promedio de **20 pedidos por hora**. ¿Cuál es la probabilidad de que, en un momento determinado al azar, el restaurante reciba **15 pedidos**?

## <font color=green>2.2 Distribución Poisson</font>
***

Se utiliza para describir el número de ocurrencias en un tiempo o espacio específico. Los eventos se caracterizan por la posibilidad de contar los éxitos, pero no poder contar los fracasos.

Como ejemplos de procesos donde podemos aplicar la distribución de Poisson tenemos la determinación del número de clientes que ingresan a una tienda en un momento determinado, el número de autos que llegan a un Auto-servicio de una cafetería a la hora del almuerzo, la determinación del número de accidentes registrados en un tramo de carretera, etc.

# $$P(k) = \frac{e^{-\mu}(\mu)^k}{k!}$$

Donde:

$e$ = constante cuyo valor aproximado es 2.718281828459045

$\mu$ = representa el número promedio de ocurrencias en un intervalo de tiempo o espacio dado

$k$ = número de éxitos en el intervalo deseado

### Experimento Poisson

1. La probabilidad de que ocurra es la misma en todo el intervalo observado.

2. El número de ocurrencias en un intervalo dado es independiente del número de ocurrencias en otros intervalos.

3. La probabilidad de que ocurra es la misma en intervalos de igual longitud.

### Media de la distribución Poisson

# $$\mu$$

### Desviación estándar de la distribución Poisson

# $$\sigma = \sqrt{\mu}$$

### Importando bibliotecas

http://www.numpy.org/

In [29]:
import numpy as np

In [30]:
#Valor de euler
np.e

2.718281828459045

## <font color='blue'>Ejemplo: Delivery</font>

Un restaurante recibe un promedio de **20 pedidos por hora**. ¿Cuál es la probabilidad de que, en un momento determinado al azar, el restaurante reciba **15 pedidos**?

### ¿Cuál es el número promedio de ocurrencias por hora ($\mu$)?

In [31]:
media = 20
media

20

### ¿Qué número de ocurrencias queremos obtener en el período ($k$)?

In [32]:
k =15
k

15

### Solución 1

In [33]:
# Solución Manual, la probabilidad de que, en un momento determinado al azar, el restaurante reciba 15 pedidos es:
probabilidad = (np.e**(-media))*(media**k)/(np.math.factorial(k))
print(f'{probabilidad}')
probabilidad

0.0516488535317584


<ipython-input-33-2f1895259851>:2: DeprecationWarning: `np.math` is a deprecated alias for the standard library `math` module (Deprecated Numpy 1.25). Replace usages of `np.math` with `math`
  probabilidad = (np.e**(-media))*(media**k)/(np.math.factorial(k))


0.0516488535317584

### Importando bibliotecas

https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.poisson.html

### Solución 2

In [34]:
# Solución con Python, la probabilidad de que, en un momento determinado al azar, el restaurante reciba 15 pedidos es:
from scipy.stats import poisson
probabilidad = poisson.pmf(k, media)
print(f'{probabilidad}')
probabilidad

0.05164885353175814


0.05164885353175814

### Ejemplo
El número media de clientes que ingresan a una panadería por hora es 20. Obtenga la probabilidad de que, en la próxima hora, ingresen exactamente 25 clientes.

In [35]:
media = 20
k =25
# Solución Manual, la probabilidad de que, en la próxima hora, ingresen exactamente 25 clientes. es:
probabilidad = poisson.pmf(k, media)
print(f'{probabilidad:.2%}')

4.46%


Lo que aprendimos en esta aula:

- Los conceptos básicos de la distribución de probabilidad Poisson
- Obtener probabilidades en problemas como los siguientes:
- Determinar la cantidad de clientes que ingresan a una tienda en un momento dado
- Número de autos que llegan al parqueo de un restaurante a la hora del almuerzo
- Determinación del número de accidentes registrados en un tramo de carretera

---
---

## <font color=red>Problema</font>
***

En un estudio sobre la altura de los residentes de una ciudad, se encontró que el conjunto de datos sigue una **distribución aproximadamente normal**, con **media 1.70** y **desviación estándar de 0.1**. Con esta información, obtenga el siguiente conjunto de probabilidades:

> **A.** probabilidad de que una persona, seleccionada al azar, sea menor de 1.80 metros.

> **B.** probabilidad de que una persona, seleccionada al azar, tenga entre 1.60 metros y 1.80 metros.

> **C.** probabilidad de que una persona, seleccionada al azar, tenga más de 1.90 metros.


## <font color=green>2.3 Distribución Normal</font>
***

La distribución normal es una de las más utilizadas en estadística. Es una distribución continua, donde la distribución de frecuencias de una variable cuantitativa tiene forma de campana y es simétrica en relación con su media.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img001.png' width='500px'>

### Características importantes

1. Es simétrica alrededor de la media;

2. El área bajo la curva corresponde a la proporción 1 o 100%;

3. Las medidas de tendencia central (media, mediana y moda) tienen el mismo valor;

4. Los extremos de la curva tienden al infinito en ambas direcciones y, teóricamente, nunca tocan el eje $x$;

5. La desviación estándar define la planitud y el ancho de la distribución. Las curvas más anchas y planas tienen valores de desviación estándar más altos;

6. La distribución se define por su media y desviación estándar;

7. La probabilidad siempre será igual al área bajo la curva, delimitada por los límites superior e inferior.

# $$f(x) = \frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Donde:

$x$ = variable normal

$\sigma$ = desviación estándar

$\mu$ = media

La probabilidad se obtiene del área bajo la curva, delimitada por los límites superior e inferior especificados. Se puede ver un ejemplo en la figura siguiente.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img002.png' width='500px'>


Para obtener el área anterior, simplemente calcule la integral de la función para los intervalos determinados. Según la siguiente ecuación:

# $$P(L_i<x<L_s) = \int_{L_i}^{L_s}\frac{1}{\sqrt{2\pi\sigma}}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^2}$$

Donde:

$x$ = variable normal

$\sigma$ = desviación estándar

$\mu$ = media

$L_i$ = límite inferior

$L_s$ = límite superior

### Tablas estandarizadas

Se crearon tablas estandarizadas para facilitar la obtención de los valores de las áreas bajo la curva normal y eliminar la necesidad de resolver integrales definidas.

Para ver los valores en una tabla estandarizada, simplemente transforme nuestra variable en una variable estandarizada $Z$.

Esta variable $Z$ representa la desviación en las desviaciones estándar de un valor de la variable original en relación con la media.

# $$Z = \frac{x-\mu}{\sigma}$$

Donde:

$x$ = variable normal con media $\mu$ y desviación estándar $\sigma$

$\sigma$ = desviación estándar

$\mu$ = media

### Construyendo la tabla normal estandarizada
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.norm.html

In [36]:
import pandas as pd
import numpy as np
from scipy.stats import norm

tabla_normal_estandarizada = pd.DataFrame(
    [],
    index=["{0:0.2f}".format(i / 100) for i in range(0, 400, 10)],
    columns = ["{0:0.2f}".format(i / 100) for i in range(0, 10)])

for index in tabla_normal_estandarizada.index:
    for column in tabla_normal_estandarizada.columns:
        Z = np.round(float(index) + float(column), 2)
        tabla_normal_estandarizada.loc[index, column] = "{0:0.4f}".format(norm.cdf(Z))

tabla_normal_estandarizada.rename_axis('Z', axis = 'columns', inplace = True)

tabla_normal_estandarizada

Z       0.00    0.01    0.02    0.03    0.04    0.05    0.06    0.07    0.08  \
0.00  0.5000  0.5040  0.5080  0.5120  0.5160  0.5199  0.5239  0.5279  0.5319   
0.10  0.5398  0.5438  0.5478  0.5517  0.5557  0.5596  0.5636  0.5675  0.5714   
0.20  0.5793  0.5832  0.5871  0.5910  0.5948  0.5987  0.6026  0.6064  0.6103   
0.30  0.6179  0.6217  0.6255  0.6293  0.6331  0.6368  0.6406  0.6443  0.6480   
0.40  0.6554  0.6591  0.6628  0.6664  0.6700  0.6736  0.6772  0.6808  0.6844   
0.50  0.6915  0.6950  0.6985  0.7019  0.7054  0.7088  0.7123  0.7157  0.7190   
0.60  0.7257  0.7291  0.7324  0.7357  0.7389  0.7422  0.7454  0.7486  0.7517   
0.70  0.7580  0.7611  0.7642  0.7673  0.7704  0.7734  0.7764  0.7794  0.7823   
0.80  0.7881  0.7910  0.7939  0.7967  0.7995  0.8023  0.8051  0.8078  0.8106   
0.90  0.8159  0.8186  0.8212  0.8238  0.8264  0.8289  0.8315  0.8340  0.8365   
1.00  0.8413  0.8438  0.8461  0.8485  0.8508  0.8531  0.8554  0.8577  0.8599   
1.10  0.8643  0.8665  0.8686  0.8708  0.8729  0.8749  0.8770  0.8790  0.8810   
1.20  0.8849  0.8869  0.8888  0.8907  0.8925  0.8944  0.8962  0.8980  0.8997   
1.30  0.9032  0.9049  0.9066  0.9082  0.9099  0.9115  0.9131  0.9147  0.9162   
1.40  0.9192  0.9207  0.9222  0.9236  0.9251  0.9265  0.9279  0.9292  0.9306   
1.50  0.9332  0.9345  0.9357  0.9370  0.9382  0.9394  0.9406  0.9418  0.9429   
1.60  0.9452  0.9463  0.9474  0.9484  0.9495  0.9505  0.9515  0.9525  0.9535   
1.70  0.9554  0.9564  0.9573  0.9582  0.9591  0.9599  0.9608  0.9616  0.9625   
1.80  0.9641  0.9649  0.9656  0.9664  0.9671  0.9678  0.9686  0.9693  0.9699   
1.90  0.9713  0.9719  0.9726  0.9732  0.9738  0.9744  0.9750  0.9756  0.9761   
2.00  0.9772  0.9778  0.9783  0.9788  0.9793  0.9798  0.9803  0.9808  0.9812   
2.10  0.9821  0.9826  0.9830  0.9834  0.9838  0.9842  0.9846  0.9850  0.9854   
2.20  0.9861  0.9864  0.9868  0.9871  0.9875  0.9878  0.9881  0.9884  0.9887   
2.30  0.9893  0.9896  0.9898  0.9901  0.9904  0.9906  0.9909  0.9911  0.9913   
2.40  0.9918  0.9920  0.9922  0.9925  0.9927  0.9929  0.9931  0.9932  0.9934   
2.50  0.9938  0.9940  0.9941  0.9943  0.9945  0.9946  0.9948  0.9949  0.9951   
2.60  0.9953  0.9955  0.9956  0.9957  0.9959  0.9960  0.9961  0.9962  0.9963   
2.70  0.9965  0.9966  0.9967  0.9968  0.9969  0.9970  0.9971  0.9972  0.9973   
2.80  0.9974  0.9975  0.9976  0.9977  0.9977  0.9978  0.9979  0.9979  0.9980   
2.90  0.9981  0.9982  0.9982  0.9983  0.9984  0.9984  0.9985  0.9985  0.9986   
3.00  0.9987  0.9987  0.9987  0.9988  0.9988  0.9989  0.9989  0.9989  0.9990   
3.10  0.9990  0.9991  0.9991  0.9991  0.9992  0.9992  0.9992  0.9992  0.9993   
3.20  0.9993  0.9993  0.9994  0.9994  0.9994  0.9994  0.9994  0.9995  0.9995   
3.30  0.9995  0.9995  0.9995  0.9996  0.9996  0.9996  0.9996  0.9996  0.9996   
3.40  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997  0.9997   
3.50  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998  0.9998   
3.60  0.9998  0.9998  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.70  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.80  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999  0.9999   
3.90  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000  1.0000   

Z       0.09  
0.00  0.5359  
0.10  0.5753  
0.20  0.6141  
0.30  0.6517  
0.40  0.6879  
0.50  0.7224  
0.60  0.7549  
0.70  0.7852  
0.80  0.8133  
0.90  0.8389  
1.00  0.8621  
1.10  0.8830  
1.20  0.9015  
1.30  0.9177  
1.40  0.9319  
1.50  0.9441  
1.60  0.9545  
1.70  0.9633  
1.80  0.9706  
1.90  0.9767  
2.00  0.9817  
2.10  0.9857  
2.20  0.9890  
2.30  0.9916  
2.40  0.9936  
2.50  0.9952  
2.60  0.9964  
2.70  0.9974  
2.80  0.9981  
2.90  0.9986  
3.00  0.9990  
3.10  0.9993  
3.20  0.9995  
3.30  0.9997  
3.40  0.9998  
3.50  0.9998  
3.60  0.9999  
3.70  0.9999  
3.80  0.9999  
3.90  1.0000

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img003.png' width='250px'>

La tabla anterior proporciona el área bajo la curva entre $-\infty$ y $Z$ desviaciones estándar por encima de la media. Recordando que dado que estos son valores estandarizados, tenemos $\mu = 0$.

## <font color='blue'>Ejemplo: ¿Cuánto mides?</font>

En un estudio sobre la altura de los residentes de una ciudad, se encontró que el conjunto de datos sigue una **distribución aproximadamente normal**, con **media 1.70** y **desviación estándar de 0.1**. Con esta información, obtenga el siguiente conjunto de probabilidades:

> **A.** probabilidad de que una persona, seleccionada al azar, mida menos que 1.80 metros.

> **B.** probabilidad de que una persona, seleccionada al azar, mida entre 1.60 metros y 1.80 metros.

> **C.** probabilidad de que una persona, seleccionada al azar, mida más de 1.90 metros.

### Problema A - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img004.png' width='350px'>


### Obtener la variable estandarizada $Z$

In [37]:
media = 1.70
media

1.7

In [38]:
desviacion_estandar = 0.1
desviacion_estandar

0.1

In [39]:
# El valor estandarizado de la probabilidad que se quiere calcular es z
z =(1.8 - media)/desviacion_estandar
z
# Este es el dato que buscamos en la Tabla Normal Estándarizada (0.8413) cuando z = 1

1.0000000000000009

### Solución 1 - Utilizando la tabla

In [40]:
probabilidad = 0.8413
probabilidad
# La probabilidad de que una persona, seleccionada al azar, mida menos que 1.80 metros es:

0.8413

### Solución 2 - Utilizando Scipy

In [41]:
from scipy.stats import norm
norm.cdf(z)
# La probabilidad de que una persona, seleccionada al azar, mida menos que 1.80 metros es:
#Este valor es más exacto ya que la Tabla trabaja condatos redondeados

0.8413447460685431

### Ejemplo: La aplicación de una prueba Estadística en una competencia presentó un conjunto de notas normalmente distribuidas. Se encontró que el conjunto de notas tenía una media de 70 y una desviación estándar de 5 puntos.¿Qué probabilidad hay de que un estudiante, seleccionado al azar, obtenga una puntuación inferior a 85 puntos?

In [42]:
media = 70
desviacion_estandar= 5
# El valor estandarizado de la probabilidad que se quiere calcular es z
z =(85 - media)/desviacion_estandar
z
norm.cdf(z)
#La probabilidad hay de que un estudiante, seleccionado al azar, obtenga una puntuación inferior a 85 puntos es:

0.9986501019683699

### Problema B - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img005.png' width='350px'>

### Obtener la variable estandarizada $Z$

In [43]:
Z =(1.70 - media)/desviacion_estandar
Z

-13.66

In [44]:
Z =(1.80 - media)/desviacion_estandar
Z

-13.64

### Solución 1 - Utilizando la tabla

In [45]:
probabilidad = (0.8413- 0.5)*2
probabilidad
# La probabilidad de que una persona, seleccionada al azar, mida entre 1.60 metros y 1.80 metros es:

0.6826000000000001

### Solución 2 - Utilizando Scipy

In [46]:
Z_superior = (1.80 - media)/desviacion_estandar
Z_superior

-13.64

In [47]:
Z_inferior = (1.60 - media)/desviacion_estandar
Z_inferior

-13.680000000000001

In [48]:
probabilidad = norm.cdf(Z_superior) - (1-norm.cdf(Z_superior))
probabilidad
# La probabilidad de que una persona, seleccionada al azar, mida entre 1.60 metros y 1.80 metros es:

-1.0

In [49]:
probabilidad = norm.cdf(Z_superior) - (norm.cdf(Z_inferior))
probabilidad
# La probabilidad de que una persona, seleccionada al azar, mida entre 1.60 metros y 1.80 metros es:

4.89339443071569e-43


### Ejemplo:

La facturación diaria de un controlador de aplicación sigue una distribución aproximadamente normal, con un media de $ 300,00 y una desviación estándar igual a $ 50,00. Obtenga las probabilidades de que, en un día aleatorio, el conductor gane:

1) Entre $ 250,00 y $ 350,00

2) Entre $ 400,00 y $ 500,00

In [50]:
from scipy.stats import norm

media = 300
desviacion_estandar = 50
Z_inferior = (250 - media) / desviacion_estandar
Z_superior = (350 - media) / desviacion_estandar

probabilidad = norm.cdf(Z_superior) - norm.cdf(Z_inferior)
print(f'{probabilidad:.2%}')

68.27%


In [51]:
from scipy.stats import norm

media = 300
desviacion_estandar = 50
Z_inferior = (400 - media) / desviacion_estandar
Z_superior = (500 - media) / desviacion_estandar

probabilidad = norm.cdf(Z_superior) - norm.cdf(Z_inferior)
print(f'{probabilidad:.2%}')

2.27%


### Problema C - Identificación del área bajo la curva

<img style='float: left' src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img006.png' width='350px'>

### Obtener la variable estandarizada $Z$

In [52]:
Z = (1.9- media) / desviacion_estandar
Z

-5.962000000000001

### Solución 1 - Utilizando la tabla

In [53]:
probabilidad =1- 0.9772 # Valor de la Tabla, se le resta a 1 para poder obtener el área a la derecha
probabilidad

0.022800000000000042

### Solución 2 - Utilizando Scipy

In [54]:
probabilidad =1- norm.cdf(Z)
probabilidad
# La diferencia se presenta porque la Tabla trabaja con valores redondeados, con python es el resultado es exacto

0.9999999987541542

In [55]:
probabilidad =norm.cdf(-Z) #Calcula el área arriba de la Z, como la normal es simetrica el valor es el mismo
probabilidad

0.9999999987541542

### Ejercicio

Una empresa verificó que las lámparas incandescentes XPTO tienen una vida útil normalmente distribuida, con un media igual a 720 días y una desviación estándar igual a 30 días. Calcula la probabilidad de que una lámpara, elegida al azar, dure:

1) Entre 650 y 750 días

2) Más de 800 días

3) Menos de 700 días

In [56]:
from scipy.stats import norm

media = 720
desviacion_estandar = 30

# Item A
Z_inferior = (650 - media) / desviacion_estandar
Z_superior = (750 - media) / desviacion_estandar

probabilidad = norm.cdf(Z_superior) - norm.cdf(Z_inferior)
print(f'{probabilidad:.2%}')

# Item B
Z = (800 - media) / desviacion_estandar

probabilidad = 1 - norm.cdf(Z)
print(f'{probabilidad:.2%}')

# Item C
Z = (700 - media) / desviacion_estandar

probabilidad = norm.cdf(Z)
print(f'{probabilidad:.2%}')

83.15%
0.38%
25.25%


### Ejercicio
Usando la tabla estandarizada, o las herramientas proporcionadas por Python, encuentre el área debajo de la curva normal para los valores Z a continuación:

1) Z < 1,96

2) Z > 2,15

3) Z < -0,78

4) Z > 0,59

In [57]:
from scipy.stats import norm

# Item A
probabilidad = norm.cdf(1.96)
print(f'{probabilidad:0.4f}')

# Item B
probabilidad = 1 - norm.cdf(2.15)
# o -> probabilidad = norm.sf(2.15)
print(f'{probabilidad:0.4f}')

# Item C
probabilidad = norm.cdf(-0.78)
print(f'{probabilidad:0.4f}')

# Item D
probabilidad = 1 - norm.cdf(0.59)
# o -> probabilidad = norm.sf(0.59)
print(f'{probabilidad:0.4f}')

0.9750
0.0158
0.2177
0.2776


Lo que aprendimos:

- Los conceptos básicos de la distribución de probabilidad Normal
- Trabajar con la tabla estandarizada Z
- Obtener las probabilidades usando la distribución normal en un conjunto de situaciones

# <font color=green>3 MUESTREO</font>
***

## <font color=green>3.1 Población y Muestra</font>
***

### Población
Conjunto de todos los elementos de interés de un estudio. Varios elementos pueden componer una población, por ejemplo: personas, edades, alturas, automóviles, etc.

En cuanto al tamaño, las poblaciones pueden ser limitadas (poblaciones finitas) o ilimitadas (poblaciones infinitas).

### Poblaciones finitas

Permiten el recuento de sus elementos. Los ejemplos incluyen el número de empleados en una empresa, el número de estudiantes en una escuela, etc.

### Poblaciones infinitas

No es posible contar sus elementos. Como ejemplos tenemos el número de porciones que se pueden extraer del agua del mar para su análisis, la temperatura medida en cada punto de un territorio, etc.

<font color=red>Cuando se pueden contar los elementos de una población, pero presentando una cantidad muy grande, se supone que la población es infinita.</font>

### Muestra
Subconjunto representativo de la población.

Los atributos numéricos de una población, como su media, varianza y desviación estándar, se conocen como **parámetros**. El enfoque principal de la inferencia estadística es precisamente generar estimaciones y probar hipótesis sobre parámetros poblacionales utilizando información de la muestra.

En resumen la muestra permite hacer inferencia sobre los parámetos

## <font color=green>3.2 ¿Cuándo usar una muestra?</font>
***

### Poblaciones infinitas

El estudio nunca llegaría a su fin. No es posible investigar todos los elementos de la población.

### Pruebas destructivas

Estudios donde los elementos evaluados son totalmente consumidos o destruidos. Ejemplo: pruebas de vida útil, pruebas de seguridad en accidentes de tráfico.

### Resultados rápidos

Investigaciones que necesitan más agilidad en la difusión. Ejemplo: encuestas de opinión, encuestas sobre problemas de salud pública.

### Altos precios

Cuando la población es finita pero muy grande, el costo de un censo puede hacer inviable el proceso.

## <font color=green>3.3 Muestreo Aleatorio Simple</font>
***

Es una de las principales formas de extraer una muestra de una población. El requisito fundamental de este tipo de enfoque es que cada elemento de la población tenga las mismas posibilidades de ser seleccionado para formar parte de la muestra.

In [58]:
# Observamos el total d eregistros u observaciones
datos.shape[0]

76840

In [59]:
#Calculamos la media de la variable Ingreso en toda la población
datos.Ingreso.mean()

2000.3831988547631

In [60]:
# Generamos una muestra de tamaño 100 y una semilla con 101 datos aleatorios
muestra = datos.sample(n = 100, random_state = 101)
muestra

Ciudad  Sexo  Edad  Color  Años de Estudio  Ingreso    Altura
29042      29     0    39      8                5      480  1.719128
62672      43     0    55      2                6      250  1.639205
29973      29     1    36      2               12      788  1.654122
22428      26     0    46      8                8     1680  1.622450
55145      41     0    37      2                9     2500  1.625268
...       ...   ...   ...    ...              ...      ...       ...
40245      32     0    35      2               12     2800  1.668898
30997      29     1    34      4               14     1924  1.795315
15094      23     0    60      4                1      130  1.682966
48788      35     1    39      2                7      460  1.777723
34891      31     0    42      2                3     1200  1.764392

[100 rows x 7 columns]

In [61]:
# Observamos que el tamaño sea 100
muestra.shape[0]

100

In [62]:
#Calculamos la media de la variable Ingreso en la muestra
muestra.Ingreso.mean()

2150.85

In [63]:
# Proporción de la Población por Sexo
datos.Sexo.value_counts(normalize = True)

Sexo
0    0.692998
1    0.307002
Name: proportion, dtype: float64

In [64]:
# Proporción de la Muestra por Sexo
muestra.Sexo.value_counts(normalize = True)

Sexo
0    0.69
1    0.31
Name: proportion, dtype: float64

Como resultado los valores de la proporción tanto en la Población como en la muestra son similares, entre mas grande el tamaño los valores de la muestra son más cercanos a los valores de la población.

## <font color=green>3.4 Muestreo Estratificado</font>
***

Es una mejora del proceso de muestreo aleatorio simple. En este método, se propone dividir la población en subgrupos de elementos con características similares, es decir, grupos más homogéneos. Con estos subgrupos separados, la técnica de muestreo aleatorio simple dentro de cada subgrupo se aplica individualmente.

## <font color=green>3.5 Muestreo por Conglomerados</font>
***

También tiene como objetivo mejorar el criterio de muestreo aleatorio simple. En el muestreo por conglomerados también se crean subgrupos, pero no serán homogéneos como en el muestreo estratificado. En el muestreo por conglomerados los subgrupos serán heterogéneos, donde luego se aplicará un muestreo aleatorio simple o estratificado.

Un ejemplo muy común de la aplicación de este tipo de técnica es la división de la población en grupos territoriales, donde los elementos investigados tendrán características bastante variadas.

Lo que aprendimos:

- Los conceptos de población y muestra
- La identificación de poblaciones finitas e infinitas
- Cuándo utilizar la técnica de muestreo en un estudio
- Técnicas de selección de muestras, como:
- Muestreo aleatorio simple
- Muestreo estratificado
- Muestreo por conglomerados

# <font color=green>4 ESTIMACIÓN</font>
***
Es la forma de obtener las estimaciones de los parametros de la Población aparatir de los parámetros de una Muestra.

## <font color='red'>Problema </font>

Suponga que los pesos de las bolsas de arroz de una industria alimentaria se distribuyen aproximadamente como una distribución normal **desviación estándar igual a 150 g**. Se seleccionó una **muestra aleatoria de 20 bolsas** de un lote específico, se obtuvo un **peso promedio de 5.050 g**. Construya un **intervalo de confianza para la media de la población** asumiendo un **nivel de significancia del 5%**.

---

Es la forma de hacer suposiciones generalizadas sobre los parámetros de una población a partir de información de una muestra.

- **Los parámetros** son los atributos numéricos de una población, como la media, la desviación estándar, etc.

- **Estimación** es el valor obtenido para un parámetro dado a partir de los datos de una muestra de la población.

## <font color=green>4.1 Teorema central del límite</font>
***

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

La desviación estándar de las medias muestrales se conoce como **error estándar de la media**

### Entendiendo el Teorema Central del Límite

In [65]:
n= 2000 # muestra
total_de_muestras = 15000

In [66]:
#Creamos un DataFrame
muestras = pd.DataFrame()
muestras

Empty DataFrame
Columns: []
Index: []

In [69]:
# Creamos un diccionario para almacenar las muestras
muestras_dict = {}

# Llenamos el diccionario con las muestras generadas
for i in range(total_de_muestras):
    muestra = datos.Edad.sample(n)
    muestra.index = range(0, len(muestra))
    muestras_dict[f'Muestra_{i}'] = muestra

# Convertimos el diccionario en un DataFrame
muestras = pd.DataFrame(muestras_dict)

# Mostramos el DataFrame de muestras
muestras

Muestra_0  Muestra_1  Muestra_2  Muestra_3  Muestra_4  Muestra_5  \
0            42         35         43         31         32         37   
1            34         59         36         44         42         20   
2            38         39         54         20         62         46   
3            48         54         34         40         57         30   
4            49         55         37         49         45         49   
...         ...        ...        ...        ...        ...        ...   
1995         34         49         54         29         64         50   
1996         29         41         55         75         51         65   
1997         47         42         45         39         73         38   
1998         39         45         62         23         59         26   
1999         30         35         61         57         52         66   

      Muestra_6  Muestra_7  Muestra_8  Muestra_9  ...  Muestra_14990  \
0            27         51         59         27  ...             32   
1            50         42         33         53  ...             19   
2            33         45         35         56  ...             44   
3            20         37         50         59  ...             66   
4            44         43         66         39  ...             39   
...         ...        ...        ...        ...  ...            ...   
1995         58         54         65         43  ...             66   
1996         22         45         31         38  ...             50   
1997         41         28         57         43  ...             54   
1998         31         66         39         37  ...             59   
1999         53         45         39         34  ...             57   

      Muestra_14991  Muestra_14992  Muestra_14993  Muestra_14994  \
0                34             49             55             30   
1                45             53             37             49   
2                42             35             31             47   
3                46             60             47             54   
4                22             22             67             28   
...             ...            ...            ...            ...   
1995             74             46             26             53   
1996             30             47             54             34   
1997             59             60             49             43   
1998             60             87             57             52   
1999             52             46             60             46   

      Muestra_14995  Muestra_14996  Muestra_14997  Muestra_14998  \
0                23             41             47             52   
1                49             54             56             70   
2                39             43             30             39   
3                50             23             44             43   
4                45             64             35             34   
...             ...            ...            ...            ...   
1995             59             68             50             57   
1996             29             52             41             60   
1997             23             41             48             39   
1998             39             36             39             43   
1999             32             51             23             31   

      Muestra_14999  
0                44  
1                40  
2                41  
3                59  
4                60  
...             ...  
1995             63  
1996             56  
1997             64  
1998             37  
1999             27  

[2000 rows x 15000 columns]

In [74]:
#Promedio de las 1500 muestras del conjunto de datos
muestras.mean()

Muestra_0        43.5305
Muestra_1        44.1860
Muestra_2        44.4490
Muestra_3        44.4360
Muestra_4        44.1790
                  ...   
Muestra_14995    44.5565
Muestra_14996    44.0835
Muestra_14997    44.1875
Muestra_14998    44.4650
Muestra_14999    44.3865
Length: 15000, dtype: float64

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

In [78]:
muestras.hist()

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

In [76]:
datos.Edad.mean()

44.07142113482561

In [77]:
datos.Edad.mean().mean()

44.07142113482561

> El **Teorema Central del Límite** establece que, con el aumento del tamaño de la muestra, la distribución de las medias muestrales se aproxima a una distribución normal con media igual a la media de la población y una desviación estándar igual a la desviación estándar de la variable original dividida por la raíz cuadrada del tamaño de la muestra. Este hecho está asegurado para $n$ mayores o iguales a 30.

# $$\sigma_\bar{x} = \frac{\sigma}{\sqrt{n}}$$

In [79]:
#Desviación estandar de la muestra
muestras.mean().std()

0.27304429415408826

In [80]:
#Desviación Estandar de la Variable Edad
datos.Edad.std()

12.480583465360187

In [81]:
# Siempre se cumple para un n > 30
datos.Edad.std()/np.sqrt(n)

0.2790743302740527

## <font color=green>4.2 Los niveles de confianza y significación</font>
***

El **nivel de confianza** ($1 - \alpha$) representa la probabilidad de que la estimación sea correcta. Además, el **nivel de significancia** ($\alpha$) expresa la probabilidad de error de la estimación.

El **nivel de confianza** representa el grado de confiabilidad del resultado estimado dentro de un cierto rango. Cuando establecemos un **nivel de confianza del 95%** en una encuesta, por ejemplo, estamos asumiendo que existe una probabilidad del 95% de que los resultados de la encuesta representen bien la realidad, es decir, sean correctos.

El **nivel de confianza** de una estimación se puede obtener del área bajo la curva normal como se muestra en la figura siguiente.

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img007.png' width='450px'>


## <font color=green>4.3 Error inferencial</font>
***

El **error inferencial** se define por la **desviación estándar de las medias muestrales** $\sigma_\bar{x}$ y por el **nivel de confianza** determinado para el proceso.

# $$e = z \frac{\sigma}{\sqrt{n}}$$

## <font color=green>4.4 Intervalos de confianza</font>
***

### Intervalo de confianza para la media poblacional

#### Con desviación estándar poblacional conocida

## $$\mu = \bar{x} \pm z\frac{\sigma}{\sqrt{n}}$$

#### Con desviación estándar de población desconocida

## $$\mu = \bar{x} \pm z\frac{s}{\sqrt{n}}$$

## <font color='blue'>Ejemplo: </font>

Suponga que los pesos de las bolsas de arroz de una industria alimentaria se distribuyen aproximadamente como una población normal con **desviación estándar igual a 150 g**. Se seleccionó una **muestra aleatoria de 20 bolsas** de un lote específico, se obtuvo un **peso promedio de 5.050 g**. Construya un intervalo de confianza para el **promedio de la población** asumiendo un **nivel de significancia del 5%**.

### Media muestral

In [82]:
media_muestra = 5050
media_muestra

5050

### Nivel de significancia ($\alpha$)

In [83]:
significancia = 0.05
significancia

0.05

### Nivel de confianza ($1 - \alpha$)

In [84]:
confianza = 1 - significancia
confianza

0.95

### Obteniendo $z$

In [85]:
tabla_normal_estandarizada[16:26]

Z       0.00    0.01    0.02    0.03    0.04    0.05    0.06    0.07    0.08  \
1.60  0.9452  0.9463  0.9474  0.9484  0.9495  0.9505  0.9515  0.9525  0.9535   
1.70  0.9554  0.9564  0.9573  0.9582  0.9591  0.9599  0.9608  0.9616  0.9625   
1.80  0.9641  0.9649  0.9656  0.9664  0.9671  0.9678  0.9686  0.9693  0.9699   
1.90  0.9713  0.9719  0.9726  0.9732  0.9738  0.9744  0.9750  0.9756  0.9761   
2.00  0.9772  0.9778  0.9783  0.9788  0.9793  0.9798  0.9803  0.9808  0.9812   
2.10  0.9821  0.9826  0.9830  0.9834  0.9838  0.9842  0.9846  0.9850  0.9854   
2.20  0.9861  0.9864  0.9868  0.9871  0.9875  0.9878  0.9881  0.9884  0.9887   
2.30  0.9893  0.9896  0.9898  0.9901  0.9904  0.9906  0.9909  0.9911  0.9913   
2.40  0.9918  0.9920  0.9922  0.9925  0.9927  0.9929  0.9931  0.9932  0.9934   
2.50  0.9938  0.9940  0.9941  0.9943  0.9945  0.9946  0.9948  0.9949  0.9951   

Z       0.09  
1.60  0.9545  
1.70  0.9633  
1.80  0.9706  
1.90  0.9767  
2.00  0.9817  
2.10  0.9857  
2.20  0.9890  
2.30  0.9916  
2.40  0.9936  
2.50  0.9952

### Obteniendo $z$

<img src='https://caelum-online-public.s3.amazonaws.com/ESP-1764-estadistica-con-python-probabilidad-y-muestreo/img008.png' width='450px'>

In [86]:
#Valor de la mitadad
0.95/2

0.475

In [87]:
# Área para buscar en la Tabla, tenemos la probabilidad y queremos encontrar el valor de z
0.5 +(0.95/2)

0.975

In [88]:
# El valor de z encontrado en la Tabla es:
1.9 +0.06

1.96

In [90]:
#Cálculo con python
z = norm.ppf(0.975)
z

1.959963984540054

### Valores de $z$ para los niveles de confianza más utilizados

|Nivel de<br>confianza|Valor del área bajo<br>la curva normal| $z$ |
|:----------------:|:---------------------------------:|:---:|
|90%               |0.95                               |1.645|
|95%               |0.975                              |1.96 |
|99%               |0.995                              |2.575|

### Obteniendo $\sigma_\bar{x}$
Obteniendo valores para calcular el Intervalo de Confianza

In [91]:
desviacion_estandar_poblacional = 150
desviacion_estandar_poblacional

150

In [92]:
n= 20
n

20

In [93]:
raiz_n = np.sqrt(n)
raiz_n

4.47213595499958

In [94]:
sigma = desviacion_estandar_poblacional / raiz_n
sigma

33.54101966249684

### Obteniendo $e$

In [96]:
# El Error es:
e= z * sigma
e

65.73919054324361

### Solución 1 - Calculando el intervalo de confianza para la media

In [98]:
# Se establecen el Limite Inferior y Superior restando y sumanta el error a la media de la muestra
intervalo = (media_muestra - e, media_muestra + e)
intervalo

(4984.260809456757, 5115.739190543243)

### Solución 2 - Calculando el intervalo de confianza para la media

In [101]:
# Calculo con Python
from scipy.stats import norm

intervalo = norm.interval(confidence=0.95, loc=media_muestra, scale=sigma)
print(intervalo)

(4984.260809456757, 5115.739190543243)


### Ejercicio
Para estimar la cantidad media gastada por cada cliente de una gran cadena de comida rápida, se seleccionó una muestra de 50 clientes.

Suponiendo que el valor de la desviación estándar de la población es de $ 6,00 y que esta población se distribuye normalmente, obtenga el margen de error de esta estimación para un nivel de confianza del 95%.

Alternativa correta
$ 0.24

In [102]:
from scipy.stats import norm
import numpy as np

z = norm.ppf(0.975)
desviacion_estandar = 6
n = 50

e = z * (desviacion_estandar / np.sqrt(n))
print(f"$ {e:0.2f}")

$ 1.66


### Ejercicio
Una muestra aleatoria simple de 1976 elementos de una población distribuida normalmente, con una desviación estándar de 11, dio como resultado una media muestral de 28.

¿Cuál es el intervalo de confianza del 90% para la media de la población?

In [105]:
from scipy.stats import norm
import numpy as np

# Datos proporcionados
media_muestral = 28
desviacion_estandar = 11
n = 1976

# Calcular el intervalo de confianza
intervalo = norm.interval(
    confidence=0.90,  # Nivel de confianza
    loc=media_muestral,
    scale=desviacion_estandar / np.sqrt(n)
)

# Imprimir el resultado
print(f"El intervalo de confianza del 90% es: {intervalo}")

El intervalo de confianza del 90% es: (27.592969951356466, 28.407030048643534)


Lo que aprendimos:

- Los conceptos de parámetros y estimaciones
- El Teorema Central de Límite
- Niveles de confianza y significancia
- Obtener el margen de error de un experimento
- Obtener intervalos de confianza para una estimación puntual (estimación por intervalo)

# <font color=green>5 CÁLCULO DEL TAMAÑO DE LA MUESTRA</font>
***

## <font color='red'>Problema </font>

Estamos estudiando los ingresos mensuales de los jefes de hogar con ingresos de hasta R$\$$ 5.000,00 en Brasil. Nuestro supervisor determinó que el **error máximo en relación a la media es de R$\$$ 10,00**. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **R$\$$ 1.082,79**. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio?

## <font color=green>5.1 Variables cuantitativas y población infinita</font>
***

# $$e = z \frac{\sigma}{\sqrt{n}}$$

#### Con desviación estándar conocida

## $$n = \left(z\frac{\sigma}{e}\right)^2$$

#### Con desviación estándar desconocida

## $$n = \left(z\frac{s}{e}\right)^2$$

Donde:

$z$ = variable normal estandarizada

$\sigma$ = desviación estándar poblacional

$s$ = desviación estándar de la muestra

$e$ = error inferencial

### <font color='red'>Comentarios</font>

1. La desviación estándar ($\sigma$ o $s$) y el error ($e$) deben estar en la misma unidad de medida.

2. Cuando el error ($e$) se representa en términos porcentuales, debe interpretarse como un porcentaje relacionado con el promedio.

## <font color='blue'>Ejemplo: Rendimiento medio</font>

Estamos estudiando los ingresos mensuales de los jefes de hogar. Nuestro supervisor determinó que el **error máximo en relación a la media es de $\$$ 100,00** pesos. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **$\$$ 3.323,39** pesos. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio?

In [106]:
# Area desde el punto medio
0.95/2

0.475

In [ ]:
# Valor de la probabilidad
0.5 + 0.95/2

In [107]:
#Cálculo con python
z = norm.ppf(0.975)
z

1.959963984540054

### Obteniendo $\sigma$

In [108]:
sigma = 3323.39

### Obteniendo $e$

In [109]:
e = 100
e

100

### Obteniendo $n$

In [116]:
# Tamaño de la muestra:
n= (z*sigma/e)**2
int(n.round())# redondeamos

4243

### Ejercicio
El valor promedio gastado por los clientes en una tienda es de $ 45.50. Suponiendo que la desviación estándar de los gastos es de $ 15.00, ¿cuál debería ser el tamaño de la muestra para estimar la media de la población con un nivel de significancia del 10%?

Tenga en cuenta que el error máximo aceptable es del 10%.

In [117]:
from scipy.stats import norm

media = 45.5
sigma = 15
significancia = 0.10
confianza = 1 - significancia

z = norm.ppf(0.5 + (confianza / 2))
error_porcentual = 0.10
e = media * error_porcentual

n = (z * (sigma / e)) ** 2
n.round()

29.0

### Ejercicio

Un fabricante de harina encontró que, en una muestra aleatoria compuesta por 200 sacos de 25 kg de un lote compuesto por 2000 sacos, mostraba una desviación estándar muestral del peso igual a 480 g.

Considerando un error máximo asociado con el promedio poblacional de 0.3 kg y un nivel de confianza del 95%, ¿qué tamaño de muestra debe seleccionarse para obtener una estimación confiable del parámetro poblacional?

In [118]:
from scipy.stats import norm

N = 2000
z = norm.ppf(0.5 + (0.95 / 2))
s = 480
e = 0.3 * 1000   # Convirtiendo kg para g

n = ((z**2) * (s**2) * (N)) / (((z**2) * (s**2)) + ((e**2) * (N - 1)))
int(n.round())

10

Lo que aprendimos:

- Determinación del tamaño de una muestra, para asegurar que sea representativa de la población
- Cálculo del tamaño de la muestra para variables cuantitativas finitas e infinitas

---
---

## <font color='red'>Problema</font>

En un lote de **10 000 latas** de gaseosa, se realizó una muestra aleatoria simple de **100 latas** y se obtuvo que la **desviación estándar muestral del contenido de las latas es igual a 12 ml**. El fabricante estipula un **error máximo en la media poblacional de solo 5 ml**. Para garantizar un **nivel de confianza del 95%**, ¿qué tamaño de muestra se debe seleccionar para este estudio?

## <font color=green>5.2 Variables cuantitativas y población finita</font>
***

#### Con desviación estándar conocida

## $$n = \frac{z^2 \sigma^2 N}{z^2 \sigma^2 + e^2(N-1)}$$

#### Con desviación estándar desconocida

## $$n = \frac{z^2 s^2 N}{z^2 s^2 + e^2(N-1)}$$

Donde:

$N$ = tamaño de la población

$z$ = variable normal estandarizada

$\sigma$ = desviación estándar poblacional

$s$ = desviación estándar muestral

$e$ = error inferencial

# <font color=green>6 FIJANDO EL CONTENIDO</font>
***

## <font color='blue'>Ejemplo: Rendimiento medio</font>

Estamos estudiando **los ingresos mensuales de los jefes de hogar con ingresos de hasta R$\$$ 5.000,00 en Brasil**. Nuestro supervisor determinó que el **error máximo en relación a la media es de R$\$$ 10,00**. Sabemos que la **desviación estándar poblacional** de este grupo de trabajadores es **R$\$$ 1.082,79** y que la **media poblacional** es **R$\$$ 1.426,54**. Para un **nivel de confianza del 95%**, ¿cuál debería ser el tamaño de la muestra de nuestro estudio? ¿Cuál es el intervalo de confianza para la media considerando el tamaño de muestra obtenido?

### Construyendo el conjunto de datos según lo especificado por el problema

In [138]:
ingresos_5000 = datos.query('Ingreso <= 5000').Ingreso
ingresos_5000

0         800
1        1150
2         880
3        3500
4         150
         ... 
76835     812
76836    1500
76837    1300
76838    1500
76839     900
Name: Ingreso, Length: 72109, dtype: int64

In [139]:
sigma= ingresos_5000.std()
sigma

1082.794549030635

In [140]:
media = ingresos_5000.mean()
media

1426.5372144947232

### Calcular el tamaño de la muestra

In [141]:
z = norm.ppf(0.975)
e =10

In [144]:
n= (z*(sigma/e))**2
int(n.round())

45039

### Calcular el intervalo de confianza para la media

In [149]:
# Hay una probabilidad del 95% de estar entre los siguientes límites:
intervalo = norm.interval(confidence=0.95, loc=media, scale=sigma/np.sqrt(n))
intervalo

(1416.5372144947232, 1436.5372144947232)